src/allmydata/node.py

```diff
@@ -211,7 +208,9 @@ def config_from_string(basedir, portnumfile, config_str, _valid_config=None):
 
     # load configuration from in-memory string
     parser = configparser.SafeConfigParser()
```

Comment:
This one doesn't use `strict=False` so it will behave differently for duplicate sections compared to `read_config`, I think.  I'm not sure why this code is factored so there need to be three different `SafeConfigParser` instantiators ...  I guess that's a problem unrelated to porting.  But I guess there should be a duplicate section test for `config_from_string` like there is one for `configutil.get_config` unless you'd rather fix the factoring to remove the repetition from the implementation.

Reply:
I removed the repetition.